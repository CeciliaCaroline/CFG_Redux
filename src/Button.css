.duneButton {
  border: 2px solid black;
  font-size: 22px;
  color: yellow;
  background-color: blue;
  font-weight: bold;
  padding: 10px;
}
