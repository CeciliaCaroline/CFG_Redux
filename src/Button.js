import './Button.css';
import React, { useState, useEffect} from 'react';

function Button(props) {
  const [currMessage, setCurrMessage] = useState("N/A");
  const [currMessageIndexSelector, setCurrMessageIndexSelector] = useState(0);

  function whenButtonPressed() {
    setCurrMessageIndexSelector(currMessageIndexSelector + 1);

    if (currMessageIndexSelector == 2){
      setCurrMessageIndexSelector(0);
    }
  }

  useEffect(() => {
    console.log("Hello World");
  }, []);

  useEffect(() => {
    const randomMessage = [
      "He who controls the spice controls the whole universe!",
      "The spice must flow",
      "Fear is the mind-killer",
    ]

    setCurrMessage(randomMessage[currMessageIndexSelector]);
  }, [currMessageIndexSelector]);

  return (
    <>
      <button
        className="duneButton"
        onClick = {whenButtonPressed}
      >
        {props.buttonText + props.appendMessage}
      </button>

      <h1>{currMessage}</h1>
    </>
  );
}

export default Button;
