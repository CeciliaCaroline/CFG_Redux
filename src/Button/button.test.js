import renderer from 'react-test-renderer';

import Button from './Button';


test('renders component correctly', () => {
    const buttonText = "123";
    const component = renderer.create(<Button buttonText={buttonText} />)
    const tree = component.toJSON();

    // creates snapshot
    expect(tree).toMatchSnapshot();
})
