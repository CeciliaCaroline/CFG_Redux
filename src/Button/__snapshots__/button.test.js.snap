// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
Array [
  <button
    className="duneButton"
    onClick={[Function]}
  >
    Output Dune messageundefined
  </button>,
  <h1>
    N/A
  </h1>,
]
`;
